src/gui/map/map_information_dialog.cpp

```diff
@@ -0,0 +1,395 @@
+	for (const auto& map_object : map_information.map_objects)
+	{
+		if (map_object.category.num)
+		{
+			addTreeItem(1, map_object.category.name, tr("%n object(s)", nullptr, map_object.category.num));
+			for (const auto& object : map_object.objects)
+			{
+				addTreeItem(2, object.name, tr("%n object(s)", nullptr, object.num));
+				if (object.colors.size())
+				{
+					for (const auto& color : object.colors)
+					{
+						addTreeItem(3, color);
+					}
+				}
+			}
+		}
+	}
+}
+
+void MapInformationDialog::addTreeItem(const int level, const QString& item, const QString& value)
+{
+	Q_ASSERT(level >= 0 && level <= 3);
+	
+	tree_items.push_back({level, item, value});
+	max_item_length = qMax(max_item_length, level * text_report_indent + item.length());
+}
+
+void MapInformationDialog::setupTreeWidget()
+{
+	for (const auto &tree_item : tree_items)
+	{
+		auto tree_depth = static_cast<int>(tree_item_hierarchy.size());
+		Q_ASSERT(tree_item.level <= tree_depth);
+	
+		for (; tree_depth > tree_item.level; --tree_depth)
+			tree_item_hierarchy.pop_back();
+	
+		QTreeWidgetItem* tree_widget_item;
+		if (tree_depth)
+			tree_widget_item = new QTreeWidgetItem(tree_item_hierarchy.back());
+		else
+			tree_widget_item = new QTreeWidgetItem(map_info_tree);
+		tree_widget_item->setText(0, tree_item.item);
+		tree_widget_item->setText(1, tree_item.value);	
+		tree_item_hierarchy.emplace_back(tree_widget_item);	// always store last tree item
+	}
+}
+
+QString MapInformationDialog::makeTextReport() const
+{
+	QString text_report;
+	for (const auto &tree_item : tree_items)
+	{
+		QString item_value;
+		if (!text_report.isEmpty() && !tree_item.level)	// separate items on topmost level
+			text_report += QChar::LineFeed;
+		item_value.fill(QChar::Space, tree_item.level * text_report_indent);
+		item_value.append(tree_item.item);
+		if (!tree_item.value.isEmpty())
+		{
+			item_value = item_value.leftJustified(max_item_length + 5, QChar::Space);
+			item_value.append(tree_item.value);
+		}
+		text_report += item_value + QChar::LineFeed;
+	}
+	return text_report;
+}
+
+// slot
+void MapInformationDialog::save()
+{
+	auto filepath = FileDialog::getSaveFileName(
+	    this,
+	    ::OpenOrienteering::MainWindow::tr("Save file"),
+	    QFileInfo(main_window->currentPath()).canonicalPath(),
+	    tr("Textfiles (*.txt)") );
+	
+	if (filepath.isEmpty())
+		return;
+	if (!filepath.endsWith(QLatin1String(".txt"), Qt::CaseInsensitive))
+		filepath.append(QLatin1String(".txt"));
+	
+	QSaveFile file(filepath);
+	if (file.open(QIODevice::WriteOnly | QIODevice::Text))
+	{
+		file.write(makeTextReport().toUtf8());
+		if (file.commit())
+			return;
+	}
+	QMessageBox::warning(this, tr("Error"),
+	                     tr("Cannot save file:\n%1\n\n%2")
+	                     .arg(filepath, file.errorString()) );
+}
+
+inline
+const QString MapInformationDialog::getFullSymbolName(const Symbol* symbol) const
```

Comment:
FWIW, clang-tidy suggests that this method can be made static.

Reply:
We probably want this to be available as a non-virtual member function `QString Symbol::getNumberAndPlainTextName() const`.